cdap-app-templates/cdap-etl/cdap-etl-proto/src/main/java/io/cdap/cdap/etl/proto/v2/ETLConfig.java

```diff
@@ -61,8 +61,8 @@
 
   protected ETLConfig(Set<ETLStage> stages, Set<Connection> connections,
                       Resources resources, Resources driverResources, Resources clientResources,
-                      boolean stageLoggingEnabled, boolean processTimingEnabled,
-                      int numOfRecordsPreview, Map<String, String> properties) {
+                      Boolean stageLoggingEnabled, Boolean processTimingEnabled,
```

Comment:
Are the values passing to the constructor allowed to be nullable? If not, you can keep using primitive type here and only change the field type to allow null values during deserialization.

Reply:
when the constructor is called by the upgrade code, can pass in these values using the getters instead of the variables directly.

Reply:
I didn't use the getter before cause we didn't want the users to confuse why {service:false} is set after upgrade when pre-upgrade it was not the case. Same for other objects, hence I made them "protected" and just pass them directly from whatever I got from serialization. 

Does that premise still hold? wdyt?

Reply:
hm ok. can you annotate these all as `@Nullable` then